src/context.js

```diff
@@ -0,0 +1,261 @@
+	 *
+	 * Usually, not to be used directly. See the static {@link module:core/context~Context.create `create()`} method.
+	 *
+	 * @param {Object} [config={}] The context config.
+	 */
+	constructor( config ) {
+		/**
+		 * Holds all configurations specific to this context instance.
+		 *
+		 * @readonly
+		 * @type {module:utils/config~Config}
+		 */
+		this.config = new Config( config );
+
+		/**
+		 * The plugins loaded and in use by this context instance.
+		 *
+		 * @readonly
+		 * @type {module:core/plugincollection~PluginCollection}
+		 */
+		this.plugins = new PluginCollection( this );
+
+		const languageConfig = this.config.get( 'language' ) || {};
+
+		/**
+		 * @readonly
+		 * @type {module:utils/locale~Locale}
+		 */
+		this.locale = new Locale( {
+			uiLanguage: typeof languageConfig === 'string' ? languageConfig : languageConfig.ui,
+			contentLanguage: this.config.get( 'language.content' )
+		} );
+
+		/**
+		 * Shorthand for {@link module:utils/locale~Locale#t}.
+		 *
+		 * @see module:utils/locale~Locale#t
+		 * @method #t
+		 */
+		this.t = this.locale.t;
+
+		/**
+		 * `true` when the context is created by an editor `false` otherwise.
+		 *
+		 * @readonly
+		 * @type {Boolean}
+		 */
+		this.wasCreatedByEditor = false;
+
+		/**
+		 * List of editors to which this context instance is injected.
+		 *
+		 * @private
+		 * @type {Set<module:core/editor/editor~Editor>}
+		 */
+		this._editors = new Set();
+	}
+
+	/**
+	 * Adds a reference to the editor which is used with this context.
+	 *
+	 * When the context is created by the editor it is additionally
+	 * marked as a {@link ~Context#wasCreatedByEditor} what is used
+	 * in the destroy chain.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor
+	 * @param {Boolean} isContextOwner
+	 */
+	addEditor( editor, isContextOwner ) {
+		if ( this.wasCreatedByEditor ) {
+			/**
+			 * Cannot add multiple editors to the context which is created by the editor.
+			 *
+			 * @error context-addEditor-private-context
+			 */
+			throw new CKEditorError(
+				'context-addEditor-private-context: Cannot add multiple editors to the context which is created by the editor.'
+			);
+		}
+
+		this._editors.add( editor );
+
+		if ( isContextOwner ) {
+			this.wasCreatedByEditor = true;
+		}
+	}
+
+	/**
+	 * Removes a reference to the editor which was used with this context.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor
+	 */
+	removeEditor( editor ) {
+		return this._editors.delete( editor );
+	}
+
+	/**
+	 * Loads and initializes plugins specified in the config.
+	 *
+	 * @returns {Promise.<module:core/plugin~LoadedPlugins>} A promise which resolves
+	 * once the initialization is completed providing an array of loaded plugins.
+	 */
+	initPlugins() {
+		const plugins = this.config.get( 'plugins' ) || [];
+
+		for ( const Plugin of plugins ) {
+			if ( typeof Plugin != 'function' ) {
+				/**
+				 * Only constructor is allowed as a Context plugin.
```

Comment:
Make this a link to the `ContextPlugin` class. The goal of those messages here is that we can extend the raw error message. On the console, we cannot link to things that easily like here.